src/components/Modal/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { cn } from "@/utils";
+import { ModalOverlayStyles } from "./index.style";
+import { ModalBackgroundStyles } from "./index.style";
+import { createContext, useContext } from "react";
+import { IoMdClose } from "react-icons/io";
+import { Button } from "../Button";
+
+import { ComponentProps } from "react";
+
+export type ModalProps = ComponentProps<"div"> & {
+  isOpen: boolean;
+  onClose: () => void;
+  closeOnOutsideClick?: boolean;
+};
+
+export type ModalTriggerProps = ComponentProps<"div"> & {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export type ModalBodyProps = ComponentProps<"div"> & {
+  children: React.ReactNode;
+};
+
+export type ModalHeaderProps = ComponentProps<"div"> & {
+  children: React.ReactNode;
+};
+
+export type ModalContentProps = ComponentProps<"div"> & {
+  children: React.ReactNode;
+};
+
+export type ModalFooterProps = ComponentProps<"div"> & {
+  children: React.ReactNode;
+};
```

Comment:
The same type seems to be repeated for a couple of types. @Suryansh-Simform I wonder if we can have a single type in this case. If in the future, we see different use cases, we'll again separate them. Thoughts?

@kashyap27102 @anjali-sim @VatsalDave2102 @RiteshAdwani @svipulc ^^ If the following sounds good to you, please feel free to refactor your components as well 😄